### DISCUSSION TITLE
V3.4.6 中文出现乱码

### DISCUSSION THREAD

**Discussion by rosyzc7** [sumatrapdfreader/sumatrapdf#3378] [Q&A]
在V3.4.6 出现中文乱码的情况
![圖片](https://user-images.githubusercontent.com/34859856/227428945-630445f5-49d3-4635-a838-d873d61f6961.png)

下面是另一个 pdf查看器
![圖片](https://user-images.githubusercontent.com/34859856/227429010-4241292c-d1d6-4de4-8db4-6b469aa6b9f6.png)

**Comment by avorobey** [comment]
Can you [open a bug](https://github.com/sumatrapdfreader/sumatrapdf/issues) and attach the PDF file which causes this?

**Comment by rosyzc7** [comment]
[Creatrol总线传感器RS485主动发码版对接协议 v2.10.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/11128735/Creatrol.RS485.v2.10.pdf)
PAGE 5 /BUG

**Comment by rosyzc7** [comment]
![圖片](https://user-images.githubusercontent.com/34859856/229272342-2d3eeb1a-39c3-44c4-be1b-4bbfd9df49a6.png)

**Comment by GitHubRulesOK** [comment]
Looks like **really bad** file CID font MojiBake (bad subsetting) https://zh.wikipedia.org/wiki/%E4%BA%82%E7%A2%BC  
https://en.wikipedia.org/wiki/Mojibake confirmed in an editor odd that its from Microsoft print to PDF (typically seen in Very output)
Most of the text in this file is images so the whole file is half baked anyway.

Here is the view in brand leader Acrobat (same as SumatraPDF)
![image](https://user-images.githubusercontent.com/12754913/229307542-573401ba-7086-4ea5-8eaa-02af6f6caf9f.png)


Personally am considering its a poor output that is not really worth fixing except **print whole file as images.**


![image](https://user-images.githubusercontent.com/12754913/229285442-0acfc84e-37f3-47f3-9574-e138fe8b8576.png)

As seen in other viewers and seen and written by MS EDGE
![image](https://user-images.githubusercontent.com/12754913/229286303-fb585a5c-2989-4bbd-b97e-15d0f0b6f191.png)

**Poor** WorkAround Print as image in Edge !
![image](https://user-images.githubusercontent.com/12754913/229289085-bdb0d777-465e-46ae-9a48-ef0a04c22109.png)

One correction is to re-set the font by naming (still not ideal, and as only a few spots use fonts such as here its not worth spending disproportionate time fixing.)
![image](https://user-images.githubusercontent.com/12754913/229289985-cc466a28-aed9-4422-8783-5a91e2ba3423.png)
Abobe reprint will "flatten" the characters, but only American is searchable.
[Creatrol.RS485.v2.10.reprint.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/11130226/Creatrol.RS485.v2.10.reprint.pdf)

![image](https://user-images.githubusercontent.com/12754913/229306915-e9043869-0810-4ac4-8d37-ded3d5e343c6.png)

**Comment by oldmii** [comment]
Has this problem been solved?
... I also encountered the same problem...
<img width="288" alt="WHIjaeAHZj" src="https://github.com/user-attachments/assets/3f0e43b8-4e0c-4394-83d0-9c0370579066" />
<img width="420" alt="YcbE6fmNcK" src="https://github.com/user-attachments/assets/ed98135f-eb8e-4959-8153-e3a6d883c931" />

**Reply by GitHubRulesOK** [reply]
@oldmii 
there are 8 billion people possible generating PDFs of different format contents and many will have one problem or another so the issues with incorrect PDF is never going to be done, An example is I am asking an AI robutt to convert a PNG to PDF which is fairly elementary as no text at all. And can it do that correctly for me with me constantly re-guiding it? Hell no I get a pure black page of potentially tens of gigabytes of rubbish carbon footprints. and have to retrain it over and over using a native command approach when I could have done it yesterday simpler using MS Edge command line print to PDF.


So my question is what does that pdf contain and did you try pre-release?

**Comment by simsq** [comment]
不同的打印机，会出现乱码。
